src/Creative/AbstractLinearCreative.php

```diff
@@ -287,11 +288,49 @@ public function addTrackingEvent($event, $url)
         
         // add event attribute
         $trackingDomElement->setAttribute('event', $event);
-        
+
+        // add offset attribute
+        if (isset($offset)) {
```

Comment:
Due to specification 2.3.2.3 "Progress Event" https://www.iab.com/wp-content/uploads/2015/06/VASTv3_0.pdf attribute offset may be applied only for progress tracking, but you add offset to method that may add adny event.

Also add to phpdoc that thi offset may be added as time value (HH:MM:SS or HH:MM:SS.mmm) or percentage (n%) value that indicates the timing for recording a view

Reply:
Thank you for feedback.

Yes you are right. The implementation with addProgressTrackingEvent is better. I will make changes.